### PR TITLE
Plugin Features

### PR DESCRIPTION
**Pull request by sjelin**
Added the option for inline plugins and pass `postTest` information about what test just ran.

Also reorganized plugin tests, as the number of configs was getting out of hand

### CODE REVIEW

File: docs/plugins.md [hunk 1]
```diff
@@ -111,11 +124,12 @@ exports.postResults = function(config) {};
  *
  * @param {Object} config The plugin configuration object.
  * @param {boolean} passed True if the test passed.
+ * @param {Object} testInfo information about the test which just ran
```
Comment (juliemr) [inline docs/plugins.md]: Style nit: capitalize Information, period after sentence.

**Comment by juliemr** [comment]
LG, just one style nit.

File: docs/plugins.md (hunk 1 above)
Comment (sjelin) [inline docs/plugins.md]: that, I can handle

**Comment by sjelin** [comment]
Closed at https://github.com/angular/protractor/commit/cf9a26f1b4bdca7d928794d24738786be074619a

**Pull request closed without merging**